### ISSUE TITLE
[Feature Request] Disable effects for specific LEDs

### ISSUE THREAD

**Issue by Haxk20** [julianschill/klipper-led_effect#61]
Currently im doing effects for my StealthBurner LEDs but it looks like when i set 2 effects that they both remain on which is fine.

It would be nice to have the ability to turn off all effects for specific LEDs like sb_leds which are 3 neopixels. And best would be to have the ability to turn off also a range of them. Like sb_leds (2-3).

For now i have to turn off all the effects that are defined for the specific LEDs. And since there is like 10 of them for only sb_leds you can imagine this is less then ideal.

**Comment by niroth81** [comment]
This would be a good addition. Even better, if part of the effect could disable all other effects on these LEDs till it's complete, then resume. For example, If I have breathe on all 20 of my LEDS, but I want leds 16-20 to be header_bed temp, then once temp is reached, disable the heater_bed effect, and go back to breathe effect.

**Comment by burntcoinpurse** [comment]
You can but you'll have to address the effect to each LED. Here's how I got mine to work. (I have quite a few pixels set up) all in series.

[display_template led_extruder_temp_glow]
text:
    {% if printer.extruder.target > 0.0 %}
        {%  set temp = printer.extruder.target %}
    {% else %}
        {% set temp = printer.configfile.settings.extruder.max_temp %}
    {% endif %}
    {% set ratio = printer.extruder.temperature / temp %}
    {ratio}, 0.0, {1-ratio}, 0.0

[display_template led_bed_temp_glow]
text:
    {% if printer.heater_bed.target > 0.0 %}
        {%  set temp = printer.heater_bed.target %}
    {% else %}
        {% set temp = printer.configfile.settings.heater_bed.max_temp %}
    {% endif %}
    {% set ratio = printer.heater_bed.temperature / temp %}
    {ratio}, 0.0, {1-ratio}, 0.0

[gcode_macro HEAT_LIGHTS_START]
gcode:
    BED_LED LED="my_led" TYPE=bed_temp MODE=glow
    HOTEND_LED LED="my_led" TYPE=extruder_temp MODE=glow

[gcode_macro HOTEND_LED]
gcode:
    {% set led = params.LED %}
    {% set type = params.TYPE %}
    {% set mode = params.MODE %}
    {% set my_neopixel = printer.configfile.settings['neopixel ' ~ led|lower] %}

    {% if type == 'clear' %}
        SET_LED_TEMPLATE LED=my_led TEMPLATE=""
        SET_LED LED=my_led RED=0 GREEN=0 BLUE=0 
    {% else %}
        {% if mode == 'glow' %}
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=16 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=17 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=18 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=19 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=20 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=21 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=22 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=23 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=24 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=25 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=26 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=27 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=28 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=29 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=30 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=31 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=32 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=33 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=34 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=35 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=36 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=37 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=38 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=39 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=40 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=41 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=42 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=43 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=44 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=45 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=46 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=47 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=48 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=49 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=50 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=51 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=52 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=53 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=54 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=55 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=56 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=57 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=58 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=59 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=60 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=61 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=62 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=63 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=64 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=65 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=66 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=67 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=68 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=69 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=70 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=134 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=135 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=136 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=137 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=138 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=139 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=140 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=141 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=142 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=143 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=144 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_extruder_temp_glow INDEX=145 TRANSMIT=0
        {% endif %}
    {% endif %}
    
[gcode_macro BED_LED]
gcode:
    {% set led = params.LED %}
    {% set type = params.TYPE %}
    {% set mode = params.MODE %}
    {% set my_neopixel = printer.configfile.settings['neopixel ' ~ led|lower] %}

    {% if type == 'clear' %}
        SET_LED_TEMPLATE LED=my_led TEMPLATE=""
        SET_LED LED=my_led RED=0 GREEN=0 BLUE=0 
    {% else %}
        {% if mode == 'glow' %}
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=1 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=2 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=84 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=85 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=86 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=87 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=88 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=89 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=90 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=91 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=92 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=93 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=94 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=95 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=96 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=97 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=98 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=99 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=100 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=101 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=102 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=103 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=104 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=105 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=106 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=107 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=108 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=109 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=110 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=111 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=112 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=113 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=114 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=115 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=116 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=117 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=118 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=119 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=120 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=121 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=122 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=123 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=124 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=125 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=126 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=127 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=128 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=129 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=130 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=131 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=132 TRANSMIT=1
            SET_LED_TEMPLATE LED=my_led TEMPLATE=led_bed_temp_glow INDEX=133 TRANSMIT=0
        {% endif %}
    {% endif %}

**Comment by burntcoinpurse** [comment]
[gcode_macro HEAT_LIGHTS_STOP]
gcode:
    BED_LED LED="my_led" TYPE=clear
    HOTEND_LED LED="my_led" TYPE=clear

You'll probably want this macro too.

**Comment by burntcoinpurse** [comment]
Niroth81 you could try making 2 macros for the breathing effect, 1 to remain on and one to clear out before starting the bed temp. Then clear the bed and restart the beathing. Only thing is, it probably won't be synced up.
